waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingFactoryImplTest.java

```diff
@@ -96,4 +97,18 @@ public void connectionLost() throws Exception {
     assertThat(mapping.getClient().get_all_databases(), is(Arrays.asList("default", "test_db")));
   }
 
+  @Test
+  public void unreachableMetastoreClient() {
+    CloseableThriftHiveMetastoreIfaceClientFactory closeableThriftHiveMetastoreIfaceClientFactory = Mockito
+        .mock(CloseableThriftHiveMetastoreIfaceClientFactory.class);
+    MetaStoreMappingFactoryImpl factory = new MetaStoreMappingFactoryImpl(prefixNamingStrategy,
+        closeableThriftHiveMetastoreIfaceClientFactory, accessControlHandlerFactory);
+    AbstractMetaStore federatedMetaStore = newFederatedInstance("fed1", thrift.getThriftConnectionUri());
+    when(closeableThriftHiveMetastoreIfaceClientFactory.newInstance(federatedMetaStore))
+        .thenThrow(new RuntimeException("Cannot create client"));
+
+    MetaStoreMapping mapping = factory.newInstance(federatedMetaStore);
+    assertThat(mapping, is(notNullValue()));
+    assertThat(mapping.isAvailable(), is(false));
+  }
```

Comment:
If the idea is that at some point the failure can be recovered from it would be good to have a test showing the sequence of events and the recovery happening.

Reply:
you need to create a new instance, that's a bit silly to test in this method

Reply:
OK, yeah, I understand the lifecycle better now.